builder/simplify.cc

```diff
@@ -217,6 +214,49 @@ expr add_constant(const expr& a, index_t b) {
   return constant_adder(b).mutate(a);
 }
 
+const_raw_buffer_ptr fold_slice_of_const_buffer(const constant_buffer& cb, const std::vector<expr>& at) {
+  bool is_slice_in_bounds = true;
+  for (size_t d = 0; d < std::min<size_t>(at.size(), cb.value->rank); ++d) {
+    if (!at[d].defined()) continue;
+    const constant* cx = at[d].as<constant>();
+    if (!cx) return nullptr;
+    if (!cb.value->dims[d].contains(cx->value)) {
+      is_slice_in_bounds = false;
+    }
+  }
+
+  // Make a buffer of the same rank as the constant, but with the sliced
+  // dimensions as singletons at the sliced `at`.
+  std::vector<dim> dims;
+  dims.reserve(cb.value->rank);
+  for (size_t d = 0; d < cb.value->rank; ++d) {
+    if (d < at.size() && at[d].defined()) {
+      const index_t v = at[d].as<constant>()->value;
+      dims.emplace_back(v, v, cb.value->dims[d].stride());
```

Comment:
Why copy the stride? That wasn't there before was it?

I don't think we should do that, that will make the constants unnecessarily large (they'll have a lot of padding).

If this was because of the out of bounds (base -> nullptr) case, I think it should be fine to just not have strides if the base is null.